enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/lock/forseti/ForsetiClient.java

```diff
@@ -697,37 +700,48 @@ else if( lock instanceof SharedLock && ((SharedLock)lock).release(this) )
     private boolean releaseLocalLock( ResourceType type, long resourceId, PrimitiveLongIntMap localLocks )
     {
         int lockCount = localLocks.remove( resourceId );
-        if(lockCount == -1)
+        if ( lockCount == -1 )
         {
             throw new IllegalStateException( this + " cannot release lock that it does not hold: " +
-                    type + "[" + resourceId + "]." );
+                                             type + "[" + resourceId + "]." );
         }
 
-        if(lockCount > 1)
+        if ( lockCount > 1 )
         {
-            localLocks.put( resourceId, lockCount-1 );
+            localLocks.put( resourceId, decrementExactToZero( lockCount ) );
             return true;
         }
         return false;
     }
 
+    private static int decrementExactToZero( int value )
+    {
+        if ( value == 0 )
```

Comment:
It's to be in line with the `incrementExact`, but we can't use `Math.decrementExact` because we need the lower bound to be zero, instead of `Integer.MIN_VALUE`. I like the symmetry, and I think the optimiser will recognise that the check can be removed. The alternative is to only guard the increments, and not the decrements.